homeassistant/components/airly/sensor.py

```diff
@@ -190,15 +188,10 @@ def __init__(
         self._attr_unique_id = (
             f"{coordinator.latitude}-{coordinator.longitude}-{description.key}".lower()
         )
+        self._attr_native_value = coordinator.data[description.key]
         self._attrs: dict[str, Any] = {}
         self.entity_description = description
 
-    @property
-    def native_value(self) -> StateType:
-        """Return the state."""
-        state = self.coordinator.data[self.entity_description.key]
-        return cast(StateType, self.entity_description.value(state))
-
     @property
     def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
Since you moved the `native_value` to the `_handle_coordinator_update`, I suggest to move this one also (just move the code as it is) would look much better.

Reply:
Currently, I'm preparing this exact change 😊 I didn't want to combine two different things in one PR.

Reply:
After some thought, I decided to refactor the state attributes. I think this will make the whole thing simpler and more readable.

Reply:
It looks much more readable 👍